code/game/machinery/martian_printer.dm

```diff
@@ -9,11 +9,17 @@
 	idle_power_usage = 20
 	active_power_usage = 5000
 	var/building = 0
+	var/cooldown_duration = 10 MINUTES
+	var/cooldown_time = 0
+	var/cooldown_active = 0
 	var/print_path = /mob/living/carbon/complex/martian
 
 /obj/machinery/mob_printer/attack_ghost(var/mob/dead/observer/O)
 	if(!canSpawn())
 		return
+	if(cooldown_active)
```

Comment:
you could totally use cooldown_time as a check here.

Reply:
I can't see a way to make the code work with at least two checks
If I only check if the timer's up, then the process() will ping every single time, because world.time is bigger than cooldown_time. Even if cooldown_time gets reset to 0 that's still the case.
Having cooldown_active be a var lets admins reset it if they want to spawn people in faster and I don't really see a problem with it, but I'm absolutely willing if there's a better way